src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -601,19 +601,33 @@ private static void PartialMethodChecks(SourceOrdinaryMethodSymbol definition, S
         {
             Debug.Assert(!ReferenceEquals(definition, implementation));
 
+            bool checkNullableMethodOverride = true;
             MethodSymbol constructedDefinition = definition.ConstructIfGeneric(implementation.TypeArgumentsWithAnnotations);
-            bool returnTypesEqual = constructedDefinition.ReturnTypeWithAnnotations.Equals(implementation.ReturnTypeWithAnnotations, TypeCompareKind.AllIgnoreOptions);
-            if (!returnTypesEqual
-                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(implementation.ReturnType)
-                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(definition.ReturnType))
+            if (!constructedDefinition.ReturnTypeWithAnnotations.Equals(implementation.ReturnTypeWithAnnotations, TypeCompareKind.AllIgnoreOptions))
             {
+                checkNullableMethodOverride = false;
                 diagnostics.Add(ErrorCode.ERR_PartialMethodReturnTypeDifference, implementation.Locations[0]);
             }
-
-            if (definition.RefKind != implementation.RefKind)
+            else if (definition.RefKind != implementation.RefKind)
             {
                 diagnostics.Add(ErrorCode.ERR_PartialMethodRefReturnDifference, implementation.Locations[0]);
             }
+            else if (!(definition.HasExplicitAccessModifier ? MemberSignatureComparer.ExtendedPartialMethodsStrictComparer : MemberSignatureComparer.PartialMethodsStrictComparer).Equals(definition, implementation))
+            {
+                checkNullableMethodOverride = false;
+                if (MemberSignatureComparer.ConsideringTupleNamesCreatesDifference(definition, implementation))
+                {
+                    diagnostics.Add(ErrorCode.ERR_PartialMethodInconsistentTupleNames, implementation.Locations[0], definition, implementation);
+                }
+                else
+                {
+                    var errorCode = definition.HasExplicitAccessModifier && MemberSignatureComparer.ExtendedPartialMethodsStrictIgnoreNullabilityComparer.Equals(definition, implementation) ?
+                        ErrorCode.ERR_PartialMethodNullabilityDifference :
```

Comment:
>ERR_PartialMethodNullabilityDifference [](start = 34, length = 38)

From our discussion, I think that nullability differences should be a warning at most, and we should ignore differences due to oblivious.